functionaltests/main_test.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+var cleanupOnFailure *bool = flag.Bool("cleanup-on-failure", true, "Whether to run cleanup even if the test failed.")
+var target *string = flag.String("target", "production", "The target environment where to run tests againts. Valid values are: qa, production")
+
+// https://www.elastic.co/guide/en/cloud/current/ec-regions-templates-instances.html
+const testRegionQA = "aws-eu-west-1"
+const testRegionProduction = "eu-west-1"
+
+// assertDocCountEqual asserts that document counts in each data stream are equal.
+func assertDocCountEqual(t *testing.T, want []esclient.ApmDocCount, actual []esclient.ApmDocCount) {
+	t.Helper()
+
+	assert.Len(t, want, len(actual))
+	for i, v := range actual {
+		assert.Equal(t, v.Count, want[i].Count, "expected doc count in %s to be equal to previous value", v.Datastream)
+	}
+}
+
+// assertDocCountGreaterThan verifies if document count in datastreams is greated than expected.
+func assertDocCountGreaterThan(t *testing.T, want []esclient.ApmDocCount, actual []esclient.ApmDocCount) {
+	t.Helper()
+
+	assert.Len(t, want, len(actual))
+	for i, v := range actual {
+		assert.Greater(t, v.Count, want[i].Count, "expected doc count in %s to be greater than previous value", v.Datastream)
+	}
+}
+
+// ingest creates and run a telemetrygen that replays multiple APM agents events to the cluster
+// a single time.
+func ingest(t *testing.T, apmURL string, apikey string) {
+	t.Helper()
+
+	t.Log("ingest data")
+	cfg := telemetrygen.DefaultConfig()
+	cfg.APIKey = apikey
+
+	u, err := url.Parse(apmURL)
+	require.NoError(t, err)
+	cfg.ServerURL = u
+
+	cfg.EventRate.Set("1000/s")
+	g, err := telemetrygen.New(cfg)
+	g.Logger = zaptest.NewLogger(t, zaptest.Level(zap.InfoLevel))
+	require.NoError(t, err)
+
+	err = g.RunBlocking(context.Background())
+}
+
+type checkDatastreamWant struct {
+	Quantity         int
+	DSManagedBy      string
+	IndicesPerDs     int
+	PreferIlm        bool
+	IndicesManagedBy []string
+}
+
+// assertDatastreams assert expected values on specific data streams in a cluster.
+func assertDatastreams(t *testing.T, expected checkDatastreamWant, actual []types.DataStream) {
+	t.Helper()
+
+	require.Len(t, actual, expected.Quantity, "number of APM datastream differs from expectations")
+	for _, v := range actual {
+		if expected.PreferIlm {
+			assert.True(t, v.PreferIlm)
+		} else {
+			assert.False(t, v.PreferIlm)
+		}
+		assert.Equal(t, expected.DSManagedBy, v.NextGenerationManagedBy.Name)
+		assert.Len(t, v.Indices, expected.IndicesPerDs)
+
+		for i, index := range v.Indices {
+			assert.Equal(t, expected.IndicesManagedBy[i], index.ManagedBy.Name)
+		}
+	}
+
+}
+
+// regionFrom returns the appropriate region to run test
+// againts based on specified target.
+func regionFrom(target string) string {
+	switch target {
+	case "qa":
+		return testRegionQA
+	case "pro":
+		return testRegionProduction
```

Comment:
I think this should be `production`, given that `target` defaults to `production`, although I see that the execution script has a `pro` abbreviation. I've never seen that one, but `prod` is widely used as a shorter version of `production`.

I think it'd be good to settle on one and avoid invalid values as defaults or valid choices.

Reply:
Settling on `pro` as that is what must be used in CI. Adding a comment to clarify this though as I understand the confusion.

Addressed in [a031acb](https://github.com/elastic/apm-server/pull/14935/commits/a031acb28504eae27b21ffe151f56cc275cf7ba1)